test/clientAuth-test.js

```diff
@@ -1,32 +1,227 @@
+var AuthorizationResponse = require('../oicMsg/oauth2/init.js').AuthorizationResponse;
+var BearerBody  = require('../src/clientAuth').BearerBody;
+var BearerHeader  = require('../src/clientAuth').BearerHeader;
+var CCAccessTokenRequest = require('../oicMsg/oauth2/init.js').CCAccessTokenRequest;
+var Client = require('../src/oauth2/init').Client;
 var CLIENT_AUTHN_METHOD = require('../src/clientAuth').CLIENT_AUTHN_METHOD;
 var ClientSecretBasic  = require('../src/clientAuth').ClientSecretBasic;
-var Client  = require('../src/oauth2/service').Client;
+var ClientSecretPost  = require('../src/clientAuth').ClientSecretPost;
+var ResourceRequest = require('../oicMsg/oauth2/init.js').ResourceRequest;
+var Service = require('../src/service').Service;
+var validClientInfo  = require('../src/clientAuth').validClientInfo;
 
-var REQ_ARGS = {'redirect_uri': 'https://example.com/rp/cb', 'response_type': "code"};
 var CLIENT_ID = "A";
 
 var CLIENT_CONF = {'issuer': 'https://example.com/as',
-               'redirect_uris': ['https://example.com/cli/authz_cb'],
-               'client_secret': 'boarding pass',
-               'client_id': CLIENT_ID};
+'redirect_uris': ['https://example.com/cli/authz_cb'],
+'client_secret': 'boarding pass',
+'client_id': CLIENT_ID};
+
+var REQ_ARGS = {'redirect_uri': 'https://example.com/rp/cb', 'response_type': "code"};
 
-function client(){
-    var client = new Client(CLIENT_AUTHN_METHOD, CLIENT_CONF);
+function getClient(){
+    var client = new Client();
+    client.init(CLIENT_AUTHN_METHOD, CLIENT_CONF);
     var sdb = client.clientInfo.stateDb;
-    sdb['ABCDE'] = {'code' : 'accessCode'};
+    sdb.dict = {};
+    sdb.dict['ABCDE'] = {'code' : 'accessCode'};
     client.clientInfo.clientSecret = 'boardingPass';
     return client;
-}
+};
+
 describe('Test client secret basic', function () {
-    var client = client();
-    var cis = client.service['accesstoken'].construct(client.clientInfo, "http://example.com", 'ABCDE');
+    var client = getClient();
+    var accessToken = new client.service['accessToken']();
+    accessToken.init();
+    var cis = accessToken.construct(client.clientInfo, {}, {'redirect_uri': 'http://example.com', 'state': 'ABCDE'});
     var csb = new ClientSecretBasic();
     var httpArgs = csb.construct(cis, client.clientInfo);
-
  
     it('test construct', function () {
-        console.log(httpArgs);
-        //assert.deepEqual(httpArgs, {'headers': {'Authorization': 'Basic QTpib2FyZGluZyBwYXNz'}});
+        var credentialsDict = {};
+        credentialsDict['A'] = 'boarding pass';
+        var authorizationDict = {};
+        authorizationDict['Authorization'] = credentialsDict;
+        var headersDict = {};
+        headersDict['headers'] = authorizationDict;
+        assert.deepEqual(headersDict, httpArgs);
+    });
+
+    it('test does not remove padding', function () {
+        var cis = new AccessTokenRequest({'code':"foo", 'redirect_uri':"http://example.com"})
```

Comment:
Always use single quotes. Add space between the key and the value (`'code': 'foo'`)